code/modules/mob/living/carbon/slime/slime.dm

```diff
@@ -634,71 +635,59 @@ mob/living/carbon/slime/var/temperature_resistance = T0C+75
-			return ..()
-		var/mob/living/simple_animal/slime/pet = new /mob/living/simple_animal/slime(M.loc)
-		pet.icon_state = "[M.colour] baby slime"
-		pet.icon_living = "[M.colour] baby slime"
-		pet.icon_dead = "[M.colour] baby slime dead"
-		pet.colour = "[M.colour]"
-		user <<"You feed the slime the potion, removing it's powers and calming it."
-		qdel(M)
-		var/newname = copytext(sanitize(input(user, "Would you like to give the slime a name?", "Name your new pet", "pet slime") as null|text),1,MAX_NAME_LEN)
-
-		if (!newname)
-			newname = "pet slime"
-		pet.name = newname
-		pet.real_name = newname
-		qdel(src)
-
-/obj/item/weapon/slimepotion2
-	name = "advanced docility potion"
-	desc = "A potent chemical mix that will nullify a slime's powers, causing it to become docile and tame. This one is meant for adult slimes"
+	desc = "A potent chemical mix that nullifies a slime's hunger, causing it to become docile and tame."
 	icon = 'icons/obj/chemical.dmi'
 	icon_state = "bottle19"
 
-/obj/item/weapon/slimepotion2/attack(mob/living/carbon/slime/M as mob, mob/user as mob)
-	if(!istype(M, /mob/living/carbon/slime/))//If target is not a slime.
-		user << "<span class='warning'> The potion only works on slimes!</span>"
+/obj/item/slimepotion/attack(mob/living/carbon/slime/M as mob, mob/user as mob)
+	if(!isslime(M))
+		user << "<span class='warning'>The potion only works on slimes!</span>"
 		return ..()
 	if(M.stat)
-		user << "<span class='warning'> The slime is dead!</span>"
+		user << "<span class='warning'>The slime is dead!</span>"
 		return..()
-	if(M.mind)
-		user << "<span class='warning'> The slime resists!</span>"
-		return ..()
-	var/mob/living/simple_animal/slime/adult/pet = new /mob/living/simple_animal/slime/adult(M.loc)
-	pet.icon_state = "[M.colour] adult slime"
-	pet.icon_living = "[M.colour] adult slime"
-	pet.icon_dead = "[M.colour] baby slime dead"
-	pet.colour = "[M.colour]"
-	user <<"You feed the slime the potion, removing it's powers and calming it."
-	qdel(M)
+
+	M.docile = 1
+	M.nutrition = 700
+	M <<"<span class='warning'> You absorb the potion and feel your intense desire to feed melt away.</span>"
+	user <<"<span class='notice'> You feed the slime the potion, removing it's hunger and calming it.</span>"
 	var/newname = copytext(sanitize(input(user, "Would you like to give the slime a name?", "Name your new pet", "pet slime") as null|text),1,MAX_NAME_LEN)
 
 	if (!newname)
 		newname = "pet slime"
-	pet.name = newname
-	pet.real_name = newname
+	M.name = newname
+	M.real_name = newname
 	qdel(src)
 
+/obj/item/slimepotion2
+	name = "sentience potion"
+	desc = "A miraculous chemical mix that can raise the intelligence of creatures to human levels."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bottle19"
+
+/obj/item/slimepotion2/afterattack(mob/living/M as mob, mob/user as mob)
+	if(!(isslime(M) || isanimal(M) || ismonkey(M) || !M.ckey)) //I'm sorry for this line
```

Comment:
My god this line is quite confusing at first.
I was about to complain about some reversed logic in this check but the entire THING is reversed, making it LOOK reversed, but actually be fine.

Jesus.